### PR TITLE
Add docs for RC2 wave of providers for 2nd round of Jan 2024

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Smaller set of providers (because reverting : 

Screenshot 2024-01-26 at 01 35 40

**Review by Taragolis — APPROVED** [review]

File: airflow/providers/microsoft/azure/CHANGELOG.rst [hunk 1]
```diff
@@ -38,6 +38,18 @@ Breaking changes
...
+   * ``Revert "Provide the logger_name param in providers hooks in order to override the logger name (#36675)" (#37015)``
+   * ``add WasbPrefixSensorTrigger params breaking change to azure provider changelog (#36940)``
+   * ``Fix stacklevel in warnings.warn into the providers (#36831)``
+   * ``Fix WasbPrefixSensor arg inconsistency between sync and async mode (#36806)``
+   * ``Fix failed tasks are not detected in 'AzureBatchHook' (#36785)``
+   * ``Fix assignment of template field in '__init__' in 'container_instances.py' (#36529)``
+   * ``Standardize airflow build process and switch to Hatchling build backend (#36537)``
+   * ``Provide the logger_name param in providers hooks in order to override the logger name (#36675)``
+   * ``Prepare docs 1st wave of Providers January 2024 (#36640)``
+   * ``Speed up autocompletion of Breeze by simplifying provider state (#36499)``
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/CHANGELOG.rst:51]: Azure provider was not ready to release in RC1 so the changelog is not edited correctly.
I suggest to drop Azure from this wave. We are waiting for another PR which is also breaking change

Comment (potiuk) [reply airflow/providers/microsoft/azure/CHANGELOG.rst:51]: I am dropping it yep - see above list :)

**Comment by potiuk** [comment]
I will send the new wave in ~ 2  hrs

**Pull request merged**